File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        Map<Character,Integer> map = new TreeMap<>();
        for(Character k : s.toCharArray())
        map.merge(k,1,Integer::sum);
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        for(Map.Entry<Character,Integer> e : map.entrySet())
        {   int k = e.getValue();
            if(k % 2 != 0 && k > max)
            max = k;
            if(k % 2 == 0 && k < min)
            min = k;
        }

        return max - min;
    }
}

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.c
int maxDifference(char* a) {
 int i,j,n,c,b[100],s,l,k;
 k=0;
 n=strlen(a);
 for(i=0;i<n;i++)
 {
    c=0;
    for(j=0;j<n;j++)
    {
    if(a[i]==a[j])
    c++;
 }
 b[k++]=c;
 }
 for(i=0;i<k;i++)
 {
    if(b[i]%2!=0)
    l=b[i];
 }
 for(i=0;i<k;i++)
 {
    if(b[i]%2!=0&&b[i]>l)
    l=b[i];
 }
  for(i=0;i<k;i++)
 {
  if(b[i]%2==0)
    s=b[i];
 }
  for(i=0;i<k;i++)
 {
    if(b[i]%2==0&&b[i]<s)
    s=b[i];
 }
 int d=l-s;
 return d;
}
